src/python/pants/backend/helm/target_types.py

```diff
@@ -105,3 +109,79 @@ class HelmChartFieldSet(FieldSet):
     chart: HelmChartMetaSourceField
     sources: HelmChartSourcesField
     dependencies: HelmChartDependenciesField
+
+
+class AllHelmChartTargets(Targets):
+    pass
+
+
+@rule
+def all_helm_chart_targets(all_targets: AllTargets) -> AllHelmChartTargets:
+    return AllHelmChartTargets([tgt for tgt in all_targets if HelmChartFieldSet.is_applicable(tgt)])
```

Comment:
This is totally fine, and common in other inference rules. But one thing to think about is that if you always have a more precise idea of where a chart should live relative to its tests, then you can also make more specific requests for targets via e.g.:
```python
# Get all targets in the same directory.
await Get(Targets, AddressSpecs([SiblingAddresses(addr.spec_path)]))
# (or other Spec types)
```

Reply:
thanks for the tip!